src/commands/basemaps-github/make.cog.github.ts

```diff
@@ -160,6 +169,96 @@ export class MakeCogGithub {
     return tileSet;
   }
 
+  /**
+   * Given a old and new lds layer stac item and log the changes for pull request body
+   */
+  getVectorChanges(newLayer: StacLink | undefined, existingLayer: StacLink | undefined): string | null {
+    // Update Layer
+    if (newLayer != null && existingLayer != null) {
+      const featureChange = Number(newLayer['lds:feature_count']) - Number(existingLayer['lds:feature_count']);
+
+      if (newLayer['lds:version'] === existingLayer['lds:version'] && featureChange !== 0) {
+        // Alert if feature changed with no version bump.
+        return `🟥🟥🟥🟥 Feature Change Detected ${newLayer['lds:name']} - version: ${newLayer['lds:version']} features: ${newLayer['lds:feature_count']} (+${featureChange}) 🟥🟥🟥🟥`;
+      }
+
+      if (featureChange >= 0) {
+        // Add Features
+        return `🟦 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} (from: ${existingLayer['lds:version']}) features: ${newLayer['lds:feature_count']} (+${featureChange})`;
+      } else {
+        // Remove Features
+        return `🟧 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} (from: ${existingLayer['lds:version']}) features: ${newLayer['lds:feature_count']} (-${featureChange})`;
+      }
+    }
+
+    // Add new Layer
+    if (newLayer != null && existingLayer == null) {
+      return `🟩 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} features: ${newLayer['lds:feature_count']}`;
+    }
+
+    // Remove Layer
+    if (newLayer == null && existingLayer != null) {
+      return `🟥 ${existingLayer['lds:name']} features: -${existingLayer['lds:feature_count']}`;
+    }
+
+    // No changes detected return null
+    return null;
+  }
+
+  /**
+   * Prepare and create pull request for the aerial tileset config
+   */
+  async diffVectorUpdate(layer: ConfigLayer, existingTileSet?: ConfigTileSetVector): Promise<string | undefined> {
+    const changes: (string | null)[] = [];
+    // Vector layer only support for 3857
+    if (layer[3857] == null) return;
+    const newCollectionPath = new URL('collection.json', layer[3857]).href;
+    const newCollection = await fsa.readJson<StacCollection>(newCollectionPath);
+    if (newCollection == null) throw new Error(`Failed to get target collection json from ${newCollectionPath}.`);
+    const ldsLayers = newCollection.links.filter((f) => f.rel === 'lds:layer') as StacLinkLds[];
+
+    // Log all the new inserts for new tileset
+    if (existingTileSet == null) {
+      changes.push(`New TileSet ts_${layer.name} with layer ${layer.name}.\n`);
+      for (const l of ldsLayers) {
+        changes.push(this.getVectorChanges(l, undefined));
+      }
+      return changes.filter((f) => f != null).join('\n');
```

Comment:
Instead of filtering here we could push only non-null items on line 224.

Reply:
I think that's okey, as we already defined the type `changes: (string | null)[]`, so we might be better to filter at end.

Reply:
The point is the code will be faster and simpler if we filter immediately rather than at the end.

Reply:
I mean, because the array allows `null`, we might accidentally adding new logic or somewhere that adds null in the array next time? It seems safer to remove this at the end. Also the maximum changes for our vector maps is about 400 lds layers. No harm to filter again here.